### ISSUE TITLE
Support retrieving valid labels/attributes for Control Measures & METOC symbols

### ISSUE THREAD

**Issue by csmoore** [Esri/joint-military-symbology-xml#122]
We need some way of determining what attributes / labels are supported for a given symbol. This seems to be defined/supported for the other symbols sets, but for these symbols sets (that have variable sets of attributes used depending on the symbol) they are not yet defined/populated in the source data/XML/sample(s).

**Comment by joebayles** [comment]
Considering that each control measure may require different labels and attributes than others in the symbol set, we'll have to implement them individually.

**Comment by csmoore** [comment]
If needed, @Dbarnes1 has put together a spreadsheet of which attributes/labels apply to which control measure - so that may help automate the process, once/if the datamodel supports.

**Comment by joebayles** [comment]
:+1:

**Comment by Dbarnes1** [comment]
Here is the [link](https://esri.box.com/s/cuvy5cpvvwjhhepbghe6g1l2aqi9gps5) on Box to the spreadsheet documenting errors with attributes/labels for Control Measure symbols, in case anyone needs it.

**Comment by joebayles** [comment]
@csmoore and/or @Dbarnes1 can you weigh in on this issue?

**Comment by csmoore** [comment]
The original idea was to create the necessary metadata and produce some output (table, tags) so we could know what attributes/labels are valid for which symbols/symbol sets. 

For instance, a UI (like the military symbol editor) we could limit/hide the attributes that don't apply.  But doesn't seem like a priority, so feel free to close.

**Comment by joebayles** [comment]
I'd like to leave this open, I think that would be useful. We can always reclose it later.

**Comment by joebayles** [comment]
So, after more review, does https://github.com/Esri/joint-military-symbology-xml/blob/dev/instance/Base.xml#L941 look like something resembling what this would be?

Basically need to come up with a way to assign an XY to it.

**Comment by abouffard** [comment]
Yes, partially.  The coordinates were my attempt at placement guidance for where that field would be placed around a typical frame.  

T on the following diagram, for example, would be drawn at relative position -1,-1
Z would be -1,-2
M would be 1,-1
G would be 1,1

![image](https://cloud.githubusercontent.com/assets/4564846/24688539/328c0fa4-1986-11e7-8683-1c1f8668be49.png)

Unfortunately, such an abstract relative position coordinate system doesn't work for control measures, so my intent was to add label rule intelligence by building on the fledgling list at https://github.com/Esri/joint-military-symbology-xml/blob/dev/instance/Base.xml#L17 and referencing those LabelRules from each control measure entity in https://github.com/Esri/joint-military-symbology-xml/blob/dev/instance/Control_Measure.xml

My design for adding label intelligence to control measure was incomplete, so there was still much to think about and build.  The complication coming from the fact that for any given control measure there can be very different labels.  You can't just map control measure labels at the symbol set level, they have to be mapped on an entity by entity basis.

**Comment by joebayles** [comment]
This is a major undertaking, but definitely doable. Should consider putting this in the backlog, but maybe commissioning database services to complete the work.

**Comment by csmoore** [comment]
Feel free to close this, we would have used this capability to only display attributes/labels valid to that symbol to the user, but I don't think we've gotten any complaints about this.

**Comment by joebayles** [comment]
We haven't received any complaints _yet_. Once people actually start using this, the requirement will be there. I'm leaving it open for now.
